### PR TITLE
[Modules][NFC] Rewrite friend-definition-2.cpp with split-file

### PR DESCRIPTION
**Pull request by ilya-biryukov**
Instead of the pragmas, which are less familiar to people.
This is a follow-up of a discussion from #111992.

### CODE REVIEW

File: clang/test/Modules/friend-definition-2.cpp [hunk 1]
```diff
@@ -1,32 +1,53 @@
...
+
+//--- b.modulemap
+module B {
+  header "b.h"
+}
+
+//--- b.h
+#ifndef B_H
+#define B_H
 template<typename T> struct ct { friend auto operator-(ct, ct) { struct X {}; return X(); } void x(); };
```
Comment (dmpolukhin) [inline clang/test/Modules/friend-definition-2.cpp:36]: I think it is better to replace it with `#include "a.h"` and it will have exactly the same issue with  #111992

**Review by dmpolukhin — APPROVED** [review]
LGTM

**Review by ChuanqiXu9 — APPROVED** [review]

File: clang/test/Modules/friend-definition-2.cpp (hunk 1 above)
Comment (ilya-biryukov) [reply clang/test/Modules/friend-definition-2.cpp:36]: Because header modules rewrite #include as imports in that case [1], this would actually change the semantics of the code.

I.e. later in `use.cc`, we would be importing from the module `A` rather than merging from the module `A` and `B`.

[1] assuming we would pass `-fmodule-map-file` and `-fmodule-file` and create a dependency chain. It's also possible to mix the same header as modular and non-modular across multiple compilations, but it quickly becomes too confusing for anyone to follow.

Comment (dmpolukhin) [reply clang/test/Modules/friend-definition-2.cpp:36]: I think in this particular case it doesn't change semantic because you don't pass `a.modulemap` when compiling `b.cpm` so it is textual inclusion. To be 100% compatible you can create a new header without modules map and textually include it in both headers (just having code duplication is not a good idea and also doesn't match what happens in reality) but I'm OK with keeping it as-is.

Comment (ilya-biryukov) [reply clang/test/Modules/friend-definition-2.cpp:36]: Yeah, that's correct, I somehow thought you meant also passing the module map. Using the same header as modular and non-modular in the same test is quite hard to follow, so I didn't want to go for it. Introducing a new textual header totally works.

Sharing the same header would've been an option, but since the file is small, hopefully it doesn't matter much.
